### ISSUE TITLE
Download slowly slows down?

### ISSUE THREAD

**Issue by Filo97** [vn-ki/anime-downloader#4]
I am on windows 10 home and the download starts at a few megabytes (about 2,5) per second but then slowly starts slowing down, usually finishind the download at around 600 kb/s. I have noticed this as an issue mainly with downloaders from the windows command line/powershell. To make sure it was just a windows problem, i tried installing it on linux, but i can't for the life of me figure out how to set it up. Could you maybe make a .sh script to install all dependencies, install anime dl, and add everything needed to PATH? That, or a very step by step tutorial would be appreciated.

**Comment by vn-ki** [comment]
I'm not sure why this happens. When I get around to a windows a machine I will try this out and verify. Does this happen with every download?

[Detailed installation](https://github.com/vn-ki/anime-downloader/wiki/Installation) instructions is available in the wiki.

Installing on linux is easy enough. If you have python and pip installed, do
```bash
pip install anime-downloader
```
If you want kissanime support,
```bash
pip install anime-downloader[cloudflare]
```
If you don't have python or pip installed, the installation depends on your distro. On ubuntu,
```bash
sudo apt-get install python3-pip
```
should do the trick.

**Comment by vn-ki** [comment]
After a quick google search, I can across this. Can you check it out and reply?

https://www.ghacks.net/2016/08/05/windows-10-limiting-internet-speed/

**Comment by Filo97** [comment]
Sorry, i was busy, however i got my hands on a linux machine and lo and behold.... It slows down. I also got a friend to test for me on THEIR linux machine. Still slows down. So it has nothing to do with windows 10.

**Comment by Filo97** [comment]
i do remember once using the script for tampermonkey to download from kissanime with IDM and downloads never slowing down. Gonna try with this right now.

**Comment by Filo97** [comment]
With aria2c it still slows down but with IDM it's constantly 5 megabytes per second from kissanime.

**Comment by Filo97** [comment]
So a HUGE difference

**Comment by Filo97** [comment]
With another download manager such as jdownloader it slows down.

**Comment by vn-ki** [comment]
Sorry for the delayed reply.

If aria2 can't download it full speed it might be hard to achieve full speed with this tool. IDM might be using multiple simultaneous connections to achieve maximum bandwidth (This can be achieved with `aria2c -s 4`, if I remember correctly), but this is a discouraged practice. 

In any case, I will investigate the slowing down.

**Comment by Filo97** [comment]
Alright, thank you very much

**Comment by Lafard** [comment]
Streaming sites generally slows down your transfer after loading the first X (say 30) seconds.

**Comment by vn-ki** [comment]
In that case, as a temporary fix maybe we can add an option to use an external downloader like aria2.

**Comment by WhoPunkedU** [comment]
Idm is a free software that is very useful when it comes to this. Although
there's some technicalities, it downloads files faster than my browser or
anime downloader. It downloads 4 files at a time at really fast speeds
without slowing down

It is a bandwidth hog though

**Comment by vn-ki** [comment]
@WhoPunkedU AFAIK, IDM does not expose an api for downloading. So it would be impossible to support it.

@Filo97 I have implemented the ability for downloading through external tools in master branch. Can you try it out and see if it fixes the download speeds for you?
**You need to have aria2 installed.**
Use the below command.
```bash
anime dl 'code geass' --external-downloader "{aria2}"
```
Replace code geass with your favourite anime :sweat_smile:. (Code geass is awesome, btw)

Installation of dev version is mentioned in the readme.

**Comment by WhoPunkedU** [comment]
I only mentioned IDM because, in my case, it is worth the extra work to get
the URLs and then rename the files as after downloading. Although it would
be cool, I wasn't suggesting to implement into the animedownloader tool.
Sorry for the confusion.

**Comment by vn-ki** [comment]
I have added partially fixed to this issue as the latest version can use external cli based downloaders.

**Comment by vn-ki** [comment]
This has been fixed in master. The newly introduced `--chunk-size` argument can be used to counter throttling.

**Comment by vn-ki** [comment]
Fixed in 3.4.0. I still recommend using aria2 for downloading.
